src/lambda/vue/samInvokeComponent.vue

```diff
@@ -1,12 +1,267 @@
+                    <select name="skipNewImageCheck" id="skipNewImageCheck" v-model="skipNewImageCheckStr">
+                        <option value="False" :key="0">False</option>
+                        <option value="True" :key="1">True</option>
+                    </select>
+                </div>
+                <div class="config-item">
+                    <label for="templateParameters">Template - Parameters</label>
+                    <input type="text" v-model="parameters.value" />
+                    <div class="json-parse-error" v-if="parameters.errorMsg">
+                        Error parsing JSON: {{ parameters.errorMsg }}
+                    </div>
+                </div>
+                <h3>api</h3>
+                <div class="config-item">
+                    <label for="querystring">Query String</label>
+                    <input type="text" v-model="launchConfig.api.querystring" />
+                </div>
+                <div class="config-item">
+                    <label for="stageVariables">Stage Variables</label>
+                    <input type="text" v-model="stageVariables.value" placeholder="Enter as valid JSON" />
+                    <div class="json-parse-error" v-if="stageVariables.errorMsg">
+                        Error parsing JSON: {{ stageVariables.errorMsg }}
+                    </div>
+                </div>
+                <div class="config-item">
+                    <label for="clientCerificateId">Client Certificate ID</label>
+                    <input type="text" v-model="launchConfig.api.clientCertificateId" />
+                </div>
+                <div class="config-item">
+                    <label for="apiPayload">API Payload</label>
+                    <input type="text" v-model="apiPayload.value" placeholder="Enter as valid JSON" />
+                    <div class="json-parse-error" v-if="apiPayload.errorMsg">
+                        Error parsing JSON: {{ apiPayload.errorMsg }}
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div class="payload-section">
+            <div class="section-header">
+                <h2>Payload</h2>
+            </div>
+            <button v-on:click.prevent="loadPayload">Load Sample Payload</button><br />
+            <textarea name="lambda-payload" id="lambda-payload" cols="60" rows="5" v-model="payload.value"></textarea>
+            <span class="data-view">payload from data: {{ payload }} </span>
+            <div class="json-parse-error" v-if="payload.errorMsg">Error parsing JSON: {{ payload.errorMsg }}</div>
+        </div>
+        <div class="invoke-button-container">
+            <button class="form-buttons" v-on:click.prevent="save">Save Debug Configuration</button>
+            <button class="form-buttons" v-on:click.prevent="launch">Invoke Debug Configuration</button>
+        </div>
+    </form>
+</template>
 
-import Vue, { VNode } from 'vue'
+<script lang="ts">
```

Comment:
Would it be easier to parse these files if we left the layout here but moved the logic to a different file? https://vuejs.org/v2/guide/single-file-components.html#What-About-Separation-of-Concerns

Reply:
Since the file is quite large I'd say yes, though the files should at least be beside each other for easy discoverability.

Reply:
I broke up the file. The type-checking still works so that was nice to see. 
Future PRs will flesh out the structure of Vue components more, and add additional type-safety for message passing. Need to refactor the remaining webviews of course.